game-api/src/index.js

```diff
@@ -23,11 +25,50 @@ function getCurrentStatus() {
 	return currentStatus;
 }
 
-const gameManager = new (require("./" + game))({
+const gameManager = new (require("./games/" + game))({
 	getCurrentStatus,
 	setStatus,
 });
 
+app.get("/backup", async (request, response) => {
+	if (await gameManager.isProcessRunning()) {
+		return response.json({"ok": false, "message": "game is running"});
+	}
+
+	const d = new Date();
+	const backupFile = game + "-backup-" + d.toISOString() + ".7z";
+
+	const stream = sevenzip.add(
+		backupFile,
+		gameManager.filesToBackup(),
+		{
+			"timeStats": true,
+			"workingDir": gameDir,
+		},
+	);
+
+	const promise = new Promise((resolve, reject) => {
+		stream.on('end', async function () {
+			const uploadErr = await gcs.uploadFile(game, backupFile);
+			if (uploadErr == null) {
```

Comment:
This plumbing may look clearer if you move the try/catch from inside `gcs.uploadFile()` to out here, rather than returning an error message/null on success.

```
try {
  await gcs.uploadFile(game, backupFile);
  console.log("done did it");
  resolve({"ok": true});
} catch (uploadErr) {
  console.warn(uploadErr);
  resolve({ok: false, error: uploadErr});
}
```

You could mark the function as `/** @throws GCloud error on failure */`, which may help IDE's warn about it.